File: algorithms/S/Stone Game IV/Stone Game IV.py
class Solution:
    def winnerSquareGame(self, n: int) -> bool:
        dyp = [False] * (n + 1)
        sqrs = []
        currSqr = 1
        for i in range(1, n + 1):
            if i == currSqr * currSqr:
                sqrs.append(i)
                currSqr += 1
                dyp[i] = True
            else:
                for sqr in sqrs:
                    if not dyp[i - sqr]:
                        dyp[i] = True
                        break
        return dyp[n]
	```
	
	### Thank You!

File: algorithms/S/Stone Game IV/Stone Game IV.cpp
class Solution {
public:
    bool winnerSquareGame(int n) {
        
        vector<int> dp(n+1,false);
        
        for(int i=1;i<=n;i++){
            for(int j=sqrt(i);j>=1;j--){
                if(dp[i-j*j] == false){
                    dp[i] = true;
                    break;
                }
            }
        }
        
        return dp[n];
    }
};
